cpp/inc/bond/core/traits.h

```diff
@@ -114,9 +114,11 @@ get_protocol_writer<Reader<Input, MarshaledBondedProtocols>, Output>
 };
 
 
+#ifdef BOND_NO_SFINAE_EXPR
```

Comment:
`check_method` is not in a detail namespace, so putting it in an `ifdef` makes this a breaking change depending on compiler version. Consider leaving it and adding a comment saying that this is no longer used by Bond with newer compilers.